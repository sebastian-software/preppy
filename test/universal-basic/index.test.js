/* global __dirname */
import { readFile } from "fs"
import { resolve } from "path"

import pify from "pify"
import rimraf from "rimraf"

import { preppy } from "../util"

const lazyRead = pify(readFile)
const lazyDelete = pify(rimraf)

test("Publish Test File via Babel as Universal", async () => {
  await lazyDelete(resolve(__dirname, "./dist"))

  await preppy({
    cwd: __dirname
  })

  expect(await lazyRead(resolve(__dirname, "dist/node.cjs.js"), "utf8")).toMatchSnapshot(
    "node-cjs"
  )
  expect(await lazyRead(resolve(__dirname, "dist/node.esm.js"), "utf8")).toMatchSnapshot(
    "node-esm"
  )
  expect(await lazyRead(resolve(__dirname, "dist/browser.esm.js"), "utf8")).toMatchSnapshot(
    "browser-esm"
  )
  expect(await lazyRead(resolve(__dirname, "dist/browser.umd.js"), "utf8")).toMatchSnapshot(
    "browser-umd"
  )
})
