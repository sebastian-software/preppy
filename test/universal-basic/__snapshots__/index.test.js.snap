// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish Test File via Babel as Universal: browser-esm 1`] = `
"/*! universal-basic v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
function browser () {
  console.log(\\"I'm in the Browser\\");
}

export { browser as default };
//# sourceMappingURL=browser.esm.js.map
"
`;

exports[`Publish Test File via Babel as Universal: browser-umd 1`] = `
"/*! universal-basic v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
  typeof define === 'function' && define.amd ? define(factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, global[\\"universal-basic\\"] = factory());
})(this, (function () { 'use strict';

  function browser () {
    console.log(\\"I'm in the Browser\\");
  }

  return browser;

}));
//# sourceMappingURL=browser.umd.js.map
"
`;

exports[`Publish Test File via Babel as Universal: node-cjs 1`] = `
"/*! universal-basic v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
'use strict';

function index () {
  console.log(\\"I'm on NodeJS\\");
}

module.exports = index;
//# sourceMappingURL=node.cjs.js.map
"
`;

exports[`Publish Test File via Babel as Universal: node-esm 1`] = `
"/*! universal-basic v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
function index () {
  console.log(\\"I'm on NodeJS\\");
}

export { index as default };
//# sourceMappingURL=node.esm.js.map
"
`;
